### PR TITLE
some fixes for when an avatar is a child of something else

### PR DESCRIPTION
**Pull request by sethalves**
- fix some bugs that appear when an avatar is a child of something else

### CODE REVIEW

**Comment by sethalves** [comment]
to test:
- with someone watching, run `script-archive/moving-platform.js` from the source repository
- it wont be perfect (avatar may pop back to origin on some frames), just better
- make sure non-parented avatar stuff in general still works

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1164a1b3bf43cba1549cc8303edf5fdf8d74d6ec available at: https://gist.github.com/fdd47bcd29fb29e6f09a1f1662bc5e56<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-1164a1b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-1164a1b.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0c2abc988342e61d54594e8447438a00bf9c801c available at: https://gist.github.com/fdd47bcd29fb29e6f09a1f1662bc5e56<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-0c2abc9.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-0c2abc9.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: bc5f563f5786c635b65aa005043c04f16c9fe818 available at: https://gist.github.com/fdd47bcd29fb29e6f09a1f1662bc5e56<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-bc5f563.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-bc5f563.dmg

File: libraries/avatars/src/AvatarData.cpp [hunk 1]
```diff
@@ -904,20 +904,16 @@ int AvatarData::parseDataFromBuffer(const QByteArray& buffer) {
 
         auto newParentID = QUuid::fromRfc4122(byteArray);
 
-        if ((_parentID != newParentID) || (_parentJointIndex = parentInfo->parentJointIndex)) {
-            _parentID = newParentID;
-            _parentJointIndex = parentInfo->parentJointIndex;
+        if ((getParentID() != newParentID) || (getParentJointIndex() != parentInfo->parentJointIndex)) {
+            SpatiallyNestable::setParentID(newParentID);
```
Comment (ZappoMan) [inline libraries/avatars/src/AvatarData.cpp]: why are you explicitly calling `SpatiallyNestable` version of this? if you call it naked, it should still call the same code, but allow for future override.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 107442eea009dd291ba6c470a758231ffa9ed59c available at: https://gist.github.com/fdd47bcd29fb29e6f09a1f1662bc5e56<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-107442e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-107442e.dmg

**Comment by Nex-Pro** [comment]
testing this

**Comment by Nex-Pro** [comment]
@sethalves  I think this is a fail.

We both observed different results.

I saw myself on a blue platform and moving with it.

Michelle saw me standing still.

When Michelle ran the script I saw her standing still and the platform moved off without her.
She saw herself moving with the platform.

each pair of images were made within a second or so.

**Me On platform:**
hifi-snap-by--on-2017-02-13_00-22-12




**Michelle on platform:**




hifi-snap-by--on-2017-02-13_00-25-32

**Comment by sethalves** [comment]
I think there is another problem (which this PR doesn't address) which is interfering with the test.  If you are both near <0,0,0> does it look any better?

**Comment by sethalves** [comment]
hm nevermind -- I see from the pictures that you were in dev-playa near the origin.  I'll try to figure out what's happening.

**Comment by sethalves** [comment]
oh!  there's a change to the server.  I should have been more thorough in the test plan.  Can you retry the test with a server running this code?

**Comment by ZappoMan** [comment]
Seth - you should bump protocol. I should have caught that in my review.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 27d55d6ff0522e0d02cac5b6d17d5a9c6a8ed0b0 available at: https://gist.github.com/fdd47bcd29fb29e6f09a1f1662bc5e56<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-27d55d6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-27d55d6.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b333126cbeed41baf24d5432aa667694848bea49 available at: https://gist.github.com/fdd47bcd29fb29e6f09a1f1662bc5e56<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-b333126.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-b333126.dmg

**Comment by OmegaHeron** [comment]
Will likely want to remove or better manage this in logs

`[02/13 15:56:30] [WARNING] [hifi.avatars] received localPosition for avatar with no parent`
`[02/13 15:56:30] [WARNING] [hifi.avatars] received localPosition for avatar with no parent`
`[02/13 15:56:31] [WARNING] [hifi.avatars] received localPosition for avatar with no parent`
`[02/13 15:56:31] [WARNING] [hifi.avatars] received localPosition for avatar with no parent`
`[02/13 15:56:32] [WARNING] [hifi.avatars] received localPosition for avatar with no parent`
`[02/13 15:56:32] [WARNING] [hifi.avatars] received localPosition for avatar with no parent`

**Comment by OmegaHeron** [comment]
Test setup
Machine 1 - Windows 10 running standard sandbox content + Interface
Machine 2 - Linux (Ubuntu 16.04.1) running Interface connecting to machine 1.
All at this PR build.

Machine 1 runs moving platform.  Machine 2 sees same(ish) as I see from local view of machine 1.
Machine 2 runs moving platform.  Same as above.
Both run moving platform.  Same as above.

This QA Pass for me with caveat of - my test is on LAN.  It may or may not work as well once exposed to running on distant servers.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3d7633a11fc4a426b52b584c28cc7b8a67c7e8c7 available at: https://gist.github.com/fdd47bcd29fb29e6f09a1f1662bc5e56<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-3d7633a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-3d7633a.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5ee484a5923ff1b1d48a9608d3d6a95cbb279bb1 available at: https://gist.github.com/fdd47bcd29fb29e6f09a1f1662bc5e56<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-5ee484a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-5ee484a.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2e2a3c60d04d714f98b4494809ca1699ca6f18a4 available at: https://gist.github.com/fdd47bcd29fb29e6f09a1f1662bc5e56<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-2e2a3c6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9652-2e2a3c6.dmg

**Comment by OmegaHeron** [comment]
Tested again since update to master and other changes, Windows 10 and Ubuntu 16.04.1

Log spam is gone.
Watching AV on moving platform from 2nd machine seems reasonable.
No issues seen from general smoke tests.

**Pull request merged**